### ISSUE TITLE
gsl::fail_fast not found in the namespace

### ISSUE THREAD

**Issue by ahmadsadeed** [microsoft/GSL#988]
I posted it as question on SO, but no answer yet:

In this simple use of C++ contracts, I get the `error: no type named 'fail_fast' in namespace 'gsl'`. Will try block throw the `fast_fail` exception or some other exception?  

```
#define GSL_THROW_ON_CONTRACT_VIOLATION
#include <gsl/gsl>
#include <iostream>


int main(void)
{
    try {
        Expects(false);
    }
    catch(const gsl::fail_fast &e) {
        std::cout << "exception: " << e.what() << '\n';
    }
}
```

**Comment by JordanMaples** [comment]
`gsl::fail_fast` and `GSL_THROW_ON_CONTRACT_VIOLATION` was removed from the GSL with #831.

A false evaluation of `Expects` or `Ensures` results in program termination.

**Comment by JordanMaples** [comment]
I also replied to your issue on [StackOverflow](https://stackoverflow.com/a/68306385/672465). I'll add the full response here that I gave on SO for completeness:

> GSL_THROW_ON_CONTRACT_VIOLATION and gsl::fast_fail were removed from the Microsoft GSL starting with release v3.0.0. All contract violations result in a call to std::terminate unless you are building in kernel mode for MSVC where it invokes __fastfail.

> There was a period of time where gsl::fast_fail was defined only when GSL_THROW_ON_CONTRACT_VIOLATION was defined, however that was identified in #267 and subsequently fixed in #268.

Context for the above point: there are comments on the OP suggesting that `gsl::fast_fail` was not being defined because `GSL_THROW_ON_CONTRACT_VIOLATION` was not defined in the correct place.
